models/references.py

```diff
@@ -85,8 +106,137 @@ class ReferenceEntry(Base, AutoIDMixin):
         doc="Any additional comments about why this reference image is bad. "
     )
 
-    # this table doesn't have provenance.
-    # The underlying image will have its own provenance for the "coaddition" process.
+    provenance_id = sa.Column(
+        sa.ForeignKey('provenances.id', ondelete="CASCADE", name='references_provenance_id_fkey'),
+        nullable=False,
+        index=True,
+        doc=(
+            "ID of the provenance of this reference. "
+            "The provenance will contain a record of the code version "
+            "and the parameters used to produce this reference. "
+        )
+    )
```

Comment:
The Reference needs a provenance so that it can keep track of the upstream provenances so it can match the Image it is linking to with the other products. 
The other option is to have it link to something other than an Image (like to the ZeroPoint or to the ImageProducts). 
If we are basing the ability of an object to find the image products on the Provenance then the reference must keep at least a list of upstream provenances (might as well save a whole Provenance at that point). 

Another thing that could be useful is that for example the reference producing pipeline might have some parameters (such as "collect all images from this date to this date") and then a Provenance will help you choose which "reference building strategy" is the one you want. Same for "custom references" that people may want to build on the fly.

Reply:
Revisiting this... as it exists right now, won't the underlying image have in its provenance all the provenances of all the derived products?  (Assuming that the code did it right and set the image's provenance from get_upstream_provenances().)  As such, I don't think the references table needs its own provenance.